contracts/protocol/facets/ProtocolInitializationFacet.sol

```diff
@@ -80,8 +82,20 @@ contract ProtocolInitializationFacet is IBosonProtocolInitializationHandler, Pro
     /**
      * @notice Initializes the version 2.2.0.
      *
+     * V2.2.0 adds the limit for the number of preminted vouchers. Cannot be initialized with ConfigHandlerFacet.initialize since it would reset the counters.
+     *
+     * @param _initializationData - data representing uint256 _maxPremintedVouchers
      */
-    function initV2_2_0() internal {}
+    function initV2_2_0(bytes calldata _initializationData) internal {
+        // v2.2.0 can only be initialized if the current version does not exist yet
+        require(protocolStatus().version == 0x0, WRONG_CURRENT_VERSION);
```

Comment:
Just to make sure that this will definitely be zero, given that it is diamond storage ? If this was a normal variable i would be comfortable, cc @zajck ?

Reply:
Version is new variable in `protocolStatus` so it's in the slot that shouldn't be occupied yet (which is true in version 2.0.0 and 2.1.0).

The only way that something was in that slot would be if
- some random data was stored there (extremely unlikely)
- we had something other variable at the end of `protocolStatus` before and we would just replace it. But since we do append only in this struct this shouldn't happen.

And this check is here more to prevent that initV2_2_0 is called later when we already initialize some higher versions.

Reply:
ok, thanks. I guess i was thinking about the probability of random date being in there, if we think that it is safe, then all good. i will OK it.